test/unit/specs/data/json-update-builder.js

```diff
@@ -0,0 +1,171 @@
+      .addArgument('$.innerArray', [4, 3, 2])
+      .create()
+
+    expect(jsonUpdateSet).to.eql({
+      '___operation': 'JSON_SET',
+      args          : {
+        '$.letter'     : 'b',
+        '$.number'     : 36,
+        '$.state'      : true,
+        '$.colours[0]' : null,
+        '$.innerObject': { a: 'b' },
+        '$.innerArray' : [4, 3, 2]
+      }
+    })
+  })
+
+  it('insert', async () => {
+    const jsonUpdateInsert = jsonUpdateBuilder.INSERT()
+
+    expect(jsonUpdateInsert.args).to.be.eql({})
+    expect(jsonUpdateInsert.operationName).to.be.equal('JSON_INSERT')
+  })
+
+  it('replace', async () => {
+    const jsonUpdateReplace = jsonUpdateBuilder.REPLACE()
+
+    expect(jsonUpdateReplace.args).to.be.eql({})
+    expect(jsonUpdateReplace.operationName).to.be.equal('JSON_REPLACE')
+  })
+
+  it('remove', async () => {
+    const jsonUpdateRemove = jsonUpdateBuilder.REMOVE()
+
+    expect(jsonUpdateRemove.args).to.be.eql([])
+    expect(jsonUpdateRemove.operationName).to.be.equal('JSON_REMOVE')
+  })
+
+  it('remove with args and run toJSON', async () => {
+    const jsonUpdateRemove = jsonUpdateBuilder.REMOVE()
+      .addArgument('$.timeMarks.date')
+      .addArgument('$.number')
+      .addArgument('$.colours[1]')
+      .toJSON()
+
+    expect(jsonUpdateRemove).to.eql({
+      '___operation': 'JSON_REMOVE',
+      args          : ['$.timeMarks.date', '$.number', '$.colours[1]']
+    })
+  })
+
+  it('array_append', async () => {
+    const jsonUpdateArrayAppend = jsonUpdateBuilder.ARRAY_APPEND()
+
+    expect(jsonUpdateArrayAppend.args).to.be.eql({})
+    expect(jsonUpdateArrayAppend.operationName).to.be.equal('JSON_ARRAY_APPEND')
+  })
+
+  it('array_insert', async () => {
+    const jsonUpdateArrayInsert = jsonUpdateBuilder.ARRAY_INSERT()
+
+    expect(jsonUpdateArrayInsert.args).to.be.eql({})
+    expect(jsonUpdateArrayInsert.operationName).to.be.equal('JSON_ARRAY_INSERT')
+  })
+
+  it('adding argument with a single value when (key,value) pair is expected for all operations except REMOVE',
+    async () => {
+      let error
+
+      try {
+        jsonUpdateBuilder.SET()
+          .addArgument('$.letter')
+          .addArgument('$.number', 36)
+          .toJSON()
+      } catch (e) {
+        error = e
+      }
+
+      expect(error.message).to.be.equal('You have to specify function\'s second argument')
+    }
+  )
+
+  const NoArgumentsErrorMessage = 'You have to add at least one argument'
+
+  it('should add at least one argument', async () => {
+    let errorSet, errorRemove
+
+    try {
+      jsonUpdateBuilder.SET().toJSON()
+    } catch (e) {
+      errorSet = e
+    }
+
+    expect(errorSet.message).to.be.equal(NoArgumentsErrorMessage)
+
+    try {
+      jsonUpdateBuilder.REMOVE().toJSON()
+    } catch (e) {
+      errorRemove = e
+    }
+
+    expect(errorRemove.message).to.be.equal(NoArgumentsErrorMessage)
+  })
+
+  it('adding arguments without setting an operation', async () => {
```

Comment:
I do not think we need this test, because this wrong usage and it's not documented 

```
 Backendless.Data.JSONUpdateBuilder
        .addArgument('$.decimals[2]', 20)
        .addArgument('$.decimals[3]', 25)
        .toJSON()
```

Reply:
+